tests/Tests.hs

```diff
@@ -0,0 +1,55 @@
+{-# LANGUAGE LambdaCase #-}
+
+import Codec.Archive.Tar.Entry (Entry (..))
+import Foliage.Tests.Tar
+import Foliage.Tests.Utils
+import System.Directory
+import System.IO
+import Test.Tasty
+import Test.Tasty.HUnit
+import Prelude hiding (EQ)
+
+main :: IO ()
+main = do
+  checkRequiredProgram "foliage"
+  defaultMain $
+    testGroup
+      "foliage-test-suite"
+      [ testCaseSteps "one" $ \step ->
+          withFixture "tests/fixtures/simple" $ do
+            step "Building repository"
+            callCommand "foliage build"
+
+            step "Running checks"
+            doesDirectoryExist "_keys" @? "_keys does not exist"
+            doesFileExist "_repo/01-index.tar" @? "01-index.tar does not exist"
+            doesFileExist "_repo/01-index.tar.gz" @? "01-index.tar.gz does not exist"
+            doesFileExist "_repo/mirrors.json" @? "mirrors.json does not exist"
+            doesFileExist "_repo/root.json" @? "root.json does not exist"
+            doesFileExist "_repo/snapshot.json" @? "snapshot.json does not exist"
+            doesFileExist "_repo/timestamp.json" @? "timestamp.json does not exist"
```

Comment:
Would it be useful to define a repository checking function in `foliage`, and then call it here? Then there's an obvious place to put these checks and update them as time goes on. Could have `foliage check` also.

Reply:
These are the files necessary for `hackage-security` they won't change.
But also another test would be to call cabal and see if it can read the repo.

Reply:
I also do think there's something to be said for: we've written some logic here that checks whether something has gone horribly wrong and we've produced a broken repository. Why would we not want to run that in the normal course of operation after building, if it's cheap?

That doesn't so obviously apply to the check for the cabal file, which is much more specific to the input data here...

Reply:
> Why would we not want to run that in the normal course of operation after building, if it's cheap?

Because I already trust what I have written in app/Foliage/CmdBuild.hs. Here I am need to write the same thing from the pov of someone who does not :) I agree this is more a unit test though. I had to start somewhere.